### PR TITLE
close resources 

### PR DESCRIPTION
**Pull request by jburel**
Review classes using ``TiffSaver(String)`` constructor and make sure the stream is closed
Tests should remain green
See https://trello.com/c/NEWizoBO/90-tiffparser-leaking-resource

### CODE REVIEW

**Comment by joshmoore** [comment]
```
[ERROR] /home/omero/workspace/OMERO-BF/bio-formats-build/bioformats/components/formats-bsd/src/loci/formats/in/OMETiffReader.java:[1343,7] cannot find symbol
```

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-DEV-merge-push#1588](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/1588/). See the [console output](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/1588/consoleText) for more details.
Possible conflicts:
  - PR #3262 melissalinkert 'OME-TIFF: rework how missing planes in TiffData are handled'
    - components/formats-bsd/src/loci/formats/in/OMETiffReader.java

--conflicts

**Comment by jburel** [comment]
The issue mentioned by @joshmoore was coming from #3262, I have temporarily excluded the PR as suggested by @melissalinkert so we can evaluate this one

File: components/formats-bsd/src/loci/formats/in/MicromanagerReader.java [hunk 1]
```diff
@@ -711,8 +710,11 @@ else if (s.length() > 100 * 1024 * 1024) {
     }
 
     byte[] b = new byte[(int) s.length()];
-    s.readFully(b);
-    s.close();
+    try {
```
Comment (sbesson) [inline components/formats-bsd/src/loci/formats/in/MicromanagerReader.java]: I assume the `RandomAccessInputStream s = new RandomAccessInputStream(jsonData);` line above could be wrapped in a more generic `try-with-resources` like other changs in the code

Comment (jburel) [reply components/formats-bsd/src/loci/formats/in/MicromanagerReader.java]: This is fixed in another PR https://github.com/openmicroscopy/bioformats/pull/3277
I was trying to split the work but maybe it is too confusing

Comment (jburel) [reply components/formats-bsd/src/loci/formats/in/MicromanagerReader.java]: I managed to create conflict during the cleaning
So probably better to combine #3277 and this one

**Comment by jburel** [comment]
To simplify the review and avoid conflicts 
 I have integrated the changes in this PR to #3277 see https://github.com/openmicroscopy/bioformats/pull/3277/commits/4b1b572c17e9d72dee721c8f7972f18235253f23

**Pull request closed without merging**